pkg/apis/networking/validation/validation.go

```diff
@@ -561,6 +568,55 @@ func validateIngressTypedLocalObjectReference(params *api.TypedLocalObjectRefere
 	return allErrs
 }
 
+// validateIngressClassParametersReference ensures that Parameters fields are valid.
+func validateIngressClassParametersReference(params *networking.IngressClassParametersReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if params == nil {
+		return allErrs
+	}
+
+	if params.APIGroup != nil {
+		for _, msg := range validation.IsDNS1123Subdomain(*params.APIGroup) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("apiGroup"), *params.APIGroup, msg))
+		}
+	}
+
+	if params.Kind == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("kind"), "kind is required"))
+	} else {
+		for _, msg := range pathvalidation.IsValidPathSegmentName(params.Kind) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), params.Kind, msg))
+		}
+	}
+
+	if params.Name == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "name is required"))
+	} else {
+		for _, msg := range pathvalidation.IsValidPathSegmentName(params.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), params.Name, msg))
+		}
+	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.IngressClassNamespacedParams) {
+		if !supportedIngressClassParametersReferenceScopes.Has(params.Scope) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("scope"), params.Scope,
+				supportedIngressClassParametersReferenceScopes.List()))
+		} else {
+			if params.Scope == networking.IngressClassParametersReferenceScopeNamespace &&
+				(params.Namespace == nil || *params.Namespace == "") {
+				allErrs = append(allErrs, field.Required(fldPath.Child("namespace"), "namespace is required when spec.parameters.scope is set to Namespace"))
+			}
+			if params.Scope == networking.IngressClassParametersReferenceScopeCluster &&
+				params.Namespace != nil {
+				allErrs = append(allErrs, field.Forbidden(fldPath.Child("namespace"), "namespace is forbidden when spec.parameters.scope is set to Cluster"))
+			}
+		}
```

Comment:
I think we're going to run into trouble here. Specifically for trying to support when this feature is enabled on a newer API server and we get an update to the one running this code. The current logic means that value would get through strategy and bypass all validation here. I think we should be validating this field anytime it is set.

I think the defaulting we're applying to scope allows us to expect this field to have one of the supported values when it is set or when the feature gate is enabled.

@thockin probably knows best as far as what's required here.

Reply:
I _think_ I see the problem you are referring to  but let me repeat what you are saying for clarity.

If the feature gate is disabled but the field is already set, then we should do the validation. This is likely possible between upgrades or when the user enables/disables the feature gate.
This then translates to executing this code-block whenever the feature gate is enabled OR whenever the field is already set.

Is that correct?

Reply:
Yep, I think that's it. Here's my understanding of how this should work:

1. We can't require a value for this field in validation unless the defaulting is also enabled.
2. The defaulting can only kick in when the feature gate is enabled.
3. If there is a value set, we need to validated it, regardless of if the feature gate is enabled.

Reply:
Okay. So I've updated the code to take this case into account and added the relevant tests for it as well.